overlord/snapstate/snapmgr.go

```diff
@@ -385,19 +385,13 @@ func (m *SnapManager) undoLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 // be changed so it looks first at the state for the snap details (Revision, Developer, etc),
 // and then complements it with information from the snap itself.
 func SnapInfo(state *state.State, snapName string, snapRevision int) (*snap.Info, error) {
-	fname := filepath.Join(dirs.SnapSnapsDir, snapName, strconv.Itoa(snapRevision), "meta", "snap.yaml")
-	yamlData, err := ioutil.ReadFile(fname)
+	fname := filepath.Join(dirs.SnapSnapsDir, snapName,
+		strconv.Itoa(snapRevision), "meta", "snap.yaml")
+	sn, err := snappy.NewInstalledSnap(fname)
 	if err != nil {
 		return nil, err
 	}
-	info, err := snap.InfoFromSnapYaml(yamlData)
-	if err != nil {
-		return nil, err
-	}
-	// Overwrite the name which doesn't belong in snap.yaml and is actually
-	// defined by snap declaration assertion.
-	// TODO: use a full SideInfo
-	info.OfficialName = snapName
-	// TODO: use state to retrieve additional information
-	return info, nil
+	snapInfo := sn.Info()
+	snapInfo.OfficialName = snapName
```

Comment:
Time to drop this dangerous hack. The proper name is coming from the store with the change above.